CapstoneProject/Assets/Scripts/Core/Puzzle/HiddenPath.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Core.Entity;
+using UnityEngine;
+
+public class HiddenPath : MonoBehaviour
+{
+    private List<Transform> childrenObject;
```

Comment:
why use a list of transforms here if all you do in each is get the component for the spriterenderer to enable or disable them all? I would just store the spriterenderers and save from doing a loop of all getcomponent each time you enter and exit.